benches/shuffle.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018-2022 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(test)]
+#![allow(non_snake_case)]
+#![feature(custom_inner_attributes)]
+#![rustfmt::skip]
+extern crate test;
+
+use test::Bencher;
+
+use core::mem::size_of;
+use rand::prelude::*;
+use rand::seq::*;
+
+// We force use of 32-bit RNG since seq code is optimised for use with 32-bit
+// generators on all platforms.
+use rand_chacha::ChaCha20Rng as CryptoRng;
+use rand_pcg::Pcg32 as SmallRng;
```

Comment:
Same as your other PR: we should not ignore perf with 64-bit generators. Also, probably makes sense to use a block RNG with less overhead like `ChaCha12Rng` (the current choice for `CryptoRng`).

Reply:
I have switched over to criterion and added benchmarks for ChaCha12 and Pcg64.  As expected, the performance improvements translate to the 64 bit rngs.